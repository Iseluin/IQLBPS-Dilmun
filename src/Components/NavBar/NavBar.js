import { useState } from 'react';
import { Link , NavLink } from 'react-router-dom';
import { FaBars } from 'react-icons/fa';
import { AiOutlineClose } from 'react-icons/ai';


import{
    HOME_ROUTE,
    CONTACT_ROUTE,
    ABOUT_ROUTE,
    SIGN_UP_ROUTE,
    SIGN_IN_ROUTE,
    LOG_OUT,
    PROFILE
} from '../../route'

function NavBar(){

     const [togglerNav, setTogglerNav] = useState(false);

     const clickHandler = () => {
         setTogglerNav(!togglerNav);
       };

    return(
            <nav className=' h-auto md:h-24 p-2 bg-blue-200 max-w-10xl mx-auto flex justify-between md:items-center'>
                <Link to={HOME_ROUTE}>
                <img src='https://cdn.discordapp.com/attachments/1031834305703460906/1035627738440159303/Asset_23.png' className='object-scale-down h-20 justify-self-center py-0 mx-4' alt='logo'/>
                </Link>
                <div
                 className={
                togglerNav ? 'flex flex-col gap-4 md:inline' : 'hidden md:inline'
                }
                >
                <NavLink className='nav-link' onClick={clickHandler} to={HOME_ROUTE} >Home</NavLink>
                <NavLink className='nav-link' onClick={clickHandler} to={CONTACT_ROUTE} >Contact</NavLink>
                <NavLink className='nav-link' onClick={clickHandler} to={ABOUT_ROUTE} >About</NavLink>
                <NavLink className='nav-link' onClick={clickHandler} to={PROFILE} >Profile</NavLink>
                <NavLink className='nav-link' onClick={clickHandler} to={SIGN_UP_ROUTE} >Sign Up</NavLink>
                <NavLink className='nav-link' onClick={clickHandler} to={SIGN_IN_ROUTE} >Sign In</NavLink>
                <NavLink className='nav-link' onClick={clickHandler} to={LOG_OUT} >Log Out</NavLink>
                </div>
                <button 
        className=' inline md:hidden self-start nav-link'
        type="button"
        onClick={clickHandler}>
        {togglerNav ? <AiOutlineClose /> : <FaBars />}
      </button>
        <button type='submit' className='mx-4'><img src='https://cdn.discordapp.com/attachments/1031834305703460906/1035608527923843122/translation-icon.png'alt='icon'/></button>
            </nav>
    )
}

export default NavBar;