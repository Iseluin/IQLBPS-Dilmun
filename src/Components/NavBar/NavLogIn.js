import { Link } from 'react-router-dom';

import{
    HOME_ROUTE,
    CONTACT_ROUTE,
    ABOUT_ROUTE,
    SIGN_UP_ROUTE,
    SIGN_IN_ROUTE,
} from '../../route'

function NavLogIn(){
    return(
            <nav>
                <Link to={HOME_ROUTE}>Home</Link>
                <Link to={CONTACT_ROUTE}>Contact</Link>
                <Link to={ABOUT_ROUTE}>About</Link>
                <Link to={SIGN_UP_ROUTE}>Sign Up</Link>
                <Link to={SIGN_IN_ROUTE}>Sign In</Link>
            </nav>
    )
}

export default NavLogIn;