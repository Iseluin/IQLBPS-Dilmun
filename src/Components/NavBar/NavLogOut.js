import { Link } from 'react-router-dom';

import{
    HOME_ROUTE,
    CONTACT_ROUTE,
    ABOUT_ROUTE,
    LOG_OUT,
    PROFILE
} from '../../route'

function NavLogIn(){
    return(
            <nav>
                <Link to={HOME_ROUTE}>Home</Link>
                <Link to={CONTACT_ROUTE}>Contact</Link>
                <Link to={ABOUT_ROUTE}>About</Link>
                <Link to={LOG_OUT}>Log Out</Link>
                <Link to={PROFILE}>Profile</Link>
            </nav>
    )
}

export default NavLogIn;