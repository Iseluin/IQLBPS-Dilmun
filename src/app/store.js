
import {
  configureStore
} from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import itemReducer from '../features/slices/item'

export const store = configureStore({
  reducer: {
    user: userReducer,
    item: itemReducer
  },
});
