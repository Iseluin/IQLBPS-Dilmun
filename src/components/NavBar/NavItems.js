import {
    AiOutlineClose
} from 'react-icons/ai';
import React, {
    useEffect,
    useState
} from 'react';
import {
    Link,
    useLocation
} from 'react-router-dom';
import {
    ABOUT_ROUTE,
    HOME_ROUTE,
    LOG_OUT,
    PRODUCT_ROUTE,
    PROFILE,
    SIGN_IN_ROUTE
} from '../../route';
import LanguageButton from '../LanguageButton/LanguageButton';

const NavItems = ({
        showItems,
        active
    }) => {

        const [open, setOpen] = useState(false);
        const location = useLocation();

        const closeNavBar = () => {
            if (open) {
                setOpen(false);
            }
        };

        useEffect(() => {
            closeNavBar();
        }, [location.key]);

  return (
    <ul className={active ? 'flex-col flex items-center fixed inset-0 left-1/4 uppercase bg-black/40 backdrop-blur-lg gap-8 justify-center p-8 md:hidden' : 'hidden'}>
        <AiOutlineClose onClick={showItems} className='text-2xl cursor-pointer'/>
        <li className='hover:text-secondary duration-300'>
            <Link to={HOME_ROUTE}>Home</Link>
            </li>
        <li className='hover:text-secondary duration-300'>
            <Link to={PRODUCT_ROUTE}>Products</Link>
            </li>
        <li className='hover:text-secondary duration-300'>
            <Link to={ABOUT_ROUTE}>About</Link>
            </li>
        <li className='hover:text-secondary duration-300'>
            <Link to={PROFILE}>Profile</Link>
            </li>
        <li className='hover:text-secondary duration-300'>
            <Link to={SIGN_IN_ROUTE}>Sign In</Link>
            </li>
        <li className='hover:text-secondary duration-300'>
            <Link to={LOG_OUT}>Log Out</Link>
            </li>
        <LanguageButton closeNavbar={closeNavBar} />
    </ul>
  );
};

export default NavItems;