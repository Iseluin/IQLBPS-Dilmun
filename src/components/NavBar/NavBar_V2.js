import {
    FaBars
} from 'react-icons/fa';
import React, {
    useEffect,
    useState
} from 'react';
import {
    NavLink,
    useLocation
} from 'react-router-dom';
// import { useDispatch, useSelector } from 'react-redux';
// import _ from 'lodash';
import NavItems from './NavItems';
import LanguageButton from '../LanguageButton/LanguageButton';
import {
    ABOUT_ROUTE,
    HOME_ROUTE,
    LOG_OUT,
    PRODUCT_ROUTE,
    PROFILE,
    SIGN_IN_ROUTE
} from '../../route';
// import { Signout, user } from '../../features/user/userSlice';

const NavBarV2 = () => {

        const [active, setActive] = useState(false)


        const showItems = () => {
            setActive(!active)
        }

        const [open, setOpen] = useState(false);
        const location = useLocation();
        // const dispatch = useDispatch();
        // const userData = useSelector(user);

        const closeNavBar = () => {
            if (open) {
                setOpen(false);
            }
        };

        useEffect(() => {
            closeNavBar();
        }, [location.key]);

  return (
    <div className='fixed w-full text-primary font-bold flex justify-between z-10 p-1 items-center bg-white/20'>
        <NavLink to={HOME_ROUTE}>
            <img
              src="https://cdn.discordapp.com/attachments/1031834305703460906/1035627738440159303/Asset_23.png"
              className="object-scale-down h-20 md:justify-self-center py-0"
              alt="logo"
            />
        </NavLink>
        <nav>
            <div className='absolute right-7 top-8 md:hidden text-4xl'>
                <FaBars onClick={showItems} className='scale-150 cursor-pointer'/>
            </div>
                <ul className='hidden md:flex gap-8 p-6 uppercase'>
                <li className='hover:text-secondary duration-300'>
                    <NavLink to={HOME_ROUTE}>Home</NavLink>
                </li>
                <li className='hover:text-secondary duration-300'>
                    <NavLink to={PRODUCT_ROUTE}>Products</NavLink>
                    </li>
                <li className='hover:text-secondary duration-300'>
                    <NavLink to={ABOUT_ROUTE}>About</NavLink>
                    </li>
                <li className='hover:text-secondary duration-300'>
                    <NavLink to={PROFILE}>Profile</NavLink>
                    </li>
                <li className='hover:text-secondary duration-300'>
                    <NavLink to={SIGN_IN_ROUTE}>Sign In</NavLink>
                    </li>
                <li className='hover:text-secondary duration-300'>
                    <NavLink to={LOG_OUT}>Log Out</NavLink>
                    </li>
                <LanguageButton closeNavbar={closeNavBar} />
                </ul>

                <NavItems showItems={showItems} active={active}/>

        </nav>

    </div>
  );
};

export default NavBarV2;