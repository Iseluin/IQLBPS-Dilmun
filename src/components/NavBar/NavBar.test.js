import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';
import { Provider } from 'react-redux';
import { store } from '../../app/store';

it('render NavBar correctly', () => {
  const { navbar } = renderer
    .create(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(navbar).toMatchSnapshot();
});
