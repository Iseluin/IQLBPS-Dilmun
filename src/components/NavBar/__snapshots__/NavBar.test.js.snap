// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render NavBar correctly 1`] = `
<nav
  class="shadow-md w-full z-10 fixed top-0 left-0 bg-primary"
>
  <div
    class="md:flex bg-primary py-2 items-center justify-between md:px-8 px-10 max-h-24"
  >
    <div>
      <a
        href="/"
      >
        <img
          alt="logo"
          class="object-scale-down h-20 md:justify-self-center py-0"
          src="https://cdn.discordapp.com/attachments/1031834305703460906/1035627738440159303/Asset_23.png"
        />
      </a>
    </div>
    <button
      type="button"
    >
      <svg
        class="text-6xl text-secondary absolute right-8 top-5 cursor-pointer md:hidden"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </button>
    <ul
      class="flex items-center md:justify-items-end justify-items-center md:flex-row flex-col md:pb-0 pb-2 absolute md:static bg-primary md:z-auto z-[-1] left-0 w-full md:w-auto transition-all duration-500 ease-in top-[-170px]"
    >
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/products"
        >
          Products
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/faq"
        >
          FAQ 
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/signup"
        >
          SignUp
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/signin"
        >
          SignIn
        </a>
      </li>
      <li>
        <a
          class="md:ml-6 text-xl md:my-0 text-background hover:text-secondary duration-500"
          href="/logout"
        >
          LogOut
        </a>
      </li>
      <div
        class="relative inline-block text-left"
        data-headlessui-state=""
        data-testid="language-button"
      >
        <div>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="md:ml-6 text-2xl md:mt-0 mt-2 text-background hover:text-secondary duration-500"
            data-headlessui-state=""
            data-testid="language-botton-toggle"
            id="headlessui-menu-button-:r0:"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title />
              <path
                d="M22.401 4.818h-9.927L10.927 0H1.599C.72 0 .002.719.002 1.599v16.275c0 .878.72 1.597 1.597 1.597h10L13.072 24H22.4c.878 0 1.597-.707 1.597-1.572V6.39c0-.865-.72-1.572-1.597-1.572zm-15.66 8.68c-2.07 0-3.75-1.68-3.75-3.75 0-2.07 1.68-3.75 3.75-3.75 1.012 0 1.86.375 2.512.976l-.99.952a2.194 2.194 0 0 0-1.522-.584c-1.305 0-2.363 1.08-2.363 2.409S5.436 12.16 6.74 12.16c1.507 0 2.13-1.08 2.19-1.808l-2.188-.002V9.066h3.51c.05.23.09.457.09.764 0 2.147-1.434 3.669-3.602 3.669zm16.757 8.93c0 .59-.492 1.072-1.097 1.072h-8.875l3.649-4.03h.005l-.74-2.302.006-.005s.568-.488 1.277-1.24c.712.771 1.63 1.699 2.818 2.805l.771-.772c-1.272-1.154-2.204-2.07-2.89-2.805.919-1.087 1.852-2.455 2.049-3.707h2.034v.002h.002v-.94h-4.532v-1.52h-1.471v1.52H14.3l-1.672-5.21.006.022h9.767c.605 0 1.097.48 1.097 1.072v16.038zm-6.484-7.311c-.536.548-.943.873-.943.873l-.008.004-1.46-4.548h4.764c-.307 1.084-.988 2.108-1.651 2.904-1.176-1.392-1.18-1.844-1.18-1.844h-1.222s.05.678 1.7 2.61z"
              />
            </svg>
          </button>
        </div>
      </div>
    </ul>
  </div>
</nav>
`;
